edge-modules/TestAnalyzer/Controllers/ReportController.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestAnalyzer.Controllers
+{
+    using System;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util.AzureLogAnalytics;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ReportController : Controller
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("Analyzer");
+
+        // GET api/report/all
+        [HttpGet("all")]
+        public async Task<string> GetReport()
+        {
+            DeviceAnalysis deviceAnalysis = Reporter.GetDeviceReport(Settings.Current.ToleranceInMilliseconds);
+            if (Settings.Current.LogAnalyticsEnabled)
+            {
+                await this.PublishToLogAnalytics(deviceAnalysis);
+            }
+
+            return deviceAnalysis.ToString();
+        }
+
+        // GET api/report (exposed for backwards compatibility for snitcher which will eventually be deprecated)
+        [HttpGet]
+        public async Task<string> GetMessages()
+        {
+            DeviceAnalysis deviceAnalysis = Reporter.GetDeviceReport(Settings.Current.ToleranceInMilliseconds);
+            if (Settings.Current.LogAnalyticsEnabled)
+            {
+                await this.PublishToLogAnalytics(deviceAnalysis); // TODO: await
+            }
+
+            return JsonConvert.SerializeObject(deviceAnalysis.MessagesReport, Formatting.Indented);
+        }
+
+        private async Task PublishToLogAnalytics(DeviceAnalysis deviceAnalysis)
+        {
+            string messagesJson = JsonConvert.SerializeObject(deviceAnalysis.MessagesReport, Formatting.Indented);
+            string twinsJson = JsonConvert.SerializeObject(deviceAnalysis.TwinsReport, Formatting.Indented);
+            string directMethodsJson = JsonConvert.SerializeObject(deviceAnalysis.DmReport, Formatting.Indented);
+
+            string workspaceId = Settings.Current.LogAnalyticsWorkspaceId;
+            string sharedKey = Settings.Current.LogAnalyticsSharedKey;
+            string logType = Settings.Current.LogAnalyticsLogType;
+
+            // Upload the data to Log Analytics for our dashboards
+            try
+            {
+                await AzureLogAnalytics.Instance.PostAsync(
+                    workspaceId,
+                    sharedKey,
+                    messagesJson,
+                    logType);
```

Comment:
should use different log type?

Reply:
I discussed this with Bear, but seeing as how the point of this PR is twin/dm testing and not log analytics, we thought it would be best to preserve the current pattern currently in master. I believe Bear has made some progress on splitting the log types.